src/components/kakaoLoginButton/useKakaoLogin.tsx

```diff
@@ -0,0 +1,50 @@
+import { useEffect } from 'react';
+import { signIn, signOut, useSession } from 'next-auth/react';
+
+import { LOCAL_STORAGE_KEY } from '~/constants/storage';
+import { post } from '~/libs/api';
+
+interface Token {
+  token_type: string;
+  access_token: string;
+}
+
+const useKakaoLogin = () => {
+  const { data: session } = useSession();
+
+  const isLoginState = !!session;
```

Comment:
```suggestion
  const { data: session, status } = useSession();

  return {status}
```

isLoginState를 만들어서 사용하는 것보다 `useSession`이 반환하는 `status`를 사용하는 것이 좋아보여요!